fpa/interfaces.go

```diff
@@ -0,0 +1,52 @@
+package fpa
+
+import (
+	"context"
+	"io"
+
+	"github.com/ipfs/go-cid"
+)
+
+// WalletManager provides access to a Lotus wallet for a Lotus node.
+type WalletManager interface {
+	NewWallet(ctx context.Context, typ string) (string, error)
+	Balance(ctx context.Context, addr string) (uint64, error)
+}
+
+// Scheduler creates and manages Job which executes Cid configurations
+// in Hot and Cold layers, enables retrieval from those layers, and
+// allows watching for Job state changes.
+type Scheduler interface {
+	Enqueue(CidConfig) (JobID, error)
```

Comment:
`EnqueueCid`?

Reply:
Yeah, sounds better. I'll also see how the _unfreeze_ action fits in the api parameters and names... since it sounds that the parameter should be of a different type of _CidConfig_.

Sounds I'd prefer avoid `Enqueue(interface{})` to accept different config types... better to have two apis than start with the type-checking world. So, a priori this change sounds good, will do in next pr.